tree/ntuple/v7/src/RField.cxx

```diff
@@ -388,6 +388,11 @@ ROOT::Experimental::Detail::RFieldBase::Create(const std::string &fieldName, con
       result = std::make_unique<RField<std::uint64_t>>(fieldName);
    } else if (canonicalType == "float") {
       result = std::make_unique<RField<float>>(fieldName);
+   } else if (canonicalType == "Float16_t") {
```

Comment:
I think we should not use `Float16_t`  for the half precision float, because in ROOT its meaning is already set to `bfloat16`.